addons/product/static/src/js/pricelist_report/product_pricelist_report.js

```diff
@@ -0,0 +1,213 @@
+    }
+
+    // events
+
+    async onClickAddQty(ev) {
+        ev.preventDefault(); // avoid automatic reloading of the page
+
+        if (this.quantities.length >= this.MAX_QTY) {
+            let message = sprintf(
+                this.env._t("At most %s quantities can be displayed simultaneously. Remove a selected quantity to add others."),
+                this.MAX_QTY
+            );
+            await this.action.doAction(sendCustomNotification("warning", message));
+            return;
+        }
+
+        const qty = parseInt($("input.add-quantity-input")[0].value);
+        if (qty && qty > 0) {
+            // Check qty already exist.
+            if (this.quantities.indexOf(qty) === -1) {
+                this.quantities.push(qty);
+                this.quantities = this.quantities.sort((a, b) => a - b);
+                this.renderHtml();
+            } else {
+                let message = sprintf(this.env._t("Quantity already present (%s)."), qty);
+                await this.action.doAction(sendCustomNotification("info", message));
+            }
+        } else {
+            await this.action.doAction(
+                sendCustomNotification("info", this.env._t("Please enter a positive whole number."))
+            );
+        }
+    }
+
+    onClickLink(ev) {
+        ev.preventDefault();
+
+        let classes = ev.target.getAttribute("class", "");
+        let resModel = ev.target.getAttribute("data-model", "");
+        let resId = ev.target.getAttribute("data-res-id", "");
+
+        if (classes && classes.includes("o_action") && resModel && resId) {
+            this.action.doAction({
+                type: 'ir.actions.act_window',
+                res_model: resModel,
+                res_id: parseInt(resId),
+                views: [[false, 'form']],
+                target: 'self',
+            });
+        }
+    }
+
+    onClickPrint() {
+        this.action.doAction({
+            type: 'ir.actions.report',
+            report_type: 'qweb-pdf',
+            report_name: 'product.report_pricelist',
+            report_file: 'product.report_pricelist',
+            data: this.reportParams,
+        });
+    }
+
+    async onClickRemoveQty(ev) {
+        if (this.quantities.length <= 1) {
+            await this.action.doAction(
+                sendCustomNotification("warning", this.env._t("You must leave at least one quantity."))
+            );
+            return;
+        }
+
+        const qty = parseInt(ev.srcElement.parentElement.childNodes[0].data);
+        this.quantities = this.quantities.filter(q => q !== qty);
+        this.renderHtml();
+    }
+
+    onSelectPricelist(ev) {
+        this.state.selectedPricelist = this.pricelists.filter(pricelist => 
+            pricelist.id === parseInt(ev.target.value)
+        )[0];
+
+        this.renderHtml();
+    }
+
+    onToggleDisplayPricelist() {
+        this.state.displayPricelistTitle = !this.displayPricelistTitle;
+        this.renderHtml();
+    }
```

Comment:
strange

Reply:
I only made getters for the properties behind the state, it avoids to have `state.property` on the xml template.